realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp

```diff
@@ -0,0 +1,209 @@
+#include <object-store/src/object_store.hpp>
```

Comment:
A general note: Instead of doing `*SR(...)` in the code, you could do `SharedRealm sr = *SR(...)` at the beginning of the method. I believe it will be easier to read.